### PR TITLE
Separate project namespace

### PR DESCRIPTION
**Pull request by bergie**
Follow-up to #700, separate project namespace (used as the library identifier when communicating with runtimes) from project ID. Allows editing the project namespace, as well as multiple checkouts of same project inside same Flowhub instance.

### CODE REVIEW

**Comment by jonnor** [comment]
@bergie will existing projects have the same `library` prefix after this change? Are there any (other) possible compatibility problems with existing projects and/or runtimes?

**Comment by bergie** [comment]
@jonnor since we fall back to `project.id` if namespace is not provided, it should remain the same

File: elements/noflo-context.html [hunk 1]
```diff
@@ -295,23 +295,11 @@
...
-        }
-        var repoParts = project.repo.split('/');
-        if (repoParts.length !== 2) {
-          return false;
-        }
-        if (library === repoParts[1]) {
-          return true;
-        }
-        if (library === repoParts[1].replace('noflo-', '')) {
+        if (library === namespace.replace('noflo-')) {
```
Comment (jonnor) [inline elements/noflo-context.html]: Why we do this?

File: elements/noflo-new-project.html [hunk 2]
```diff
@@ -72,6 +74,7 @@
         this.fire('new', {
           id: this.projectId.replace(/[\/\s\.]/g, '-'),
           name: this.name,
+          namespace: this.namespace.replace(/[\/\s\.]/g, '-'),
```
Comment (jonnor) [inline elements/noflo-new-project.html]: Shouldn't this already be fine from the replacements in `nameChanged`? And why are the replacement logics there different?

File: elements/noflo-project-inspector.html [hunk 3]
```diff
@@ -25,6 +25,10 @@
           <input type="text" value="{{ name }}" placeholder="My Cool Project" required>
         </label>
         <label>
+          <span>Project namespace</span>
```
Comment (jonnor) [inline elements/noflo-project-inspector.html]: Should explain that this is the component library name/prefix.

File: elements/noflo-new-project.html (hunk 2 above)
Comment (bergie) [inline elements/noflo-new-project.html]: Dealt with

**Pull request merged**